config/routes.rb

```diff
@@ -16,13 +16,21 @@
 
   namespace :api do
     namespace :v1 do
+      namespace :admin do
+        namespace :users do
+          get '/search_all', to: 'search_all#index'
+        end
+        patch '/roles/:id',   to: 'roles#update'
+        delete '/roles/:id',  to: 'roles#destroy'
```

Comment:
We could probably also use `resources :role, only: [:update, :delete]`, which is a little less verbose. Thoughts?